### ISSUE TITLE
Discovery example doesn't build

### ISSUE THREAD

**Issue by gdesmott** [fxbox/multicast-dns#9]
Trying to build the discovery example from `README`:
```
error[E0412]: cannot find type `ServiceInfo` in this scope
 --> examples/discovery.rs:9:41
  |
9 |     let on_service_resolved = |service: ServiceInfo| {
  |                                         ^^^^^^^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
  |
2 | use multicast_dns::discovery::discovery_manager::ServiceInfo;
  |

error[E0412]: cannot find type `ServiceInfo` in this scope
  --> examples/discovery.rs:13:43
   |
13 |     let on_service_discovered = |service: ServiceInfo| {
   |                                           ^^^^^^^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
2  | use multicast_dns::discovery::discovery_manager::ServiceInfo;
   |

error[E0422]: cannot find struct, variant or union type `ResolveListeners` in this scope
  --> examples/discovery.rs:16:33
   |
16 |         let resolve_listeners = ResolveListeners {
   |                                 ^^^^^^^^^^^^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
2  | use multicast_dns::discovery::discovery_manager::ResolveListeners;
   |

error[E0422]: cannot find struct, variant or union type `DiscoveryListeners` in this scope
  --> examples/discovery.rs:27:31
   |
27 |     let discovery_listeners = DiscoveryListeners {
   |                               ^^^^^^^^^^^^^^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
2  | use multicast_dns::discovery::discovery_manager::DiscoveryListeners;
   |
```

Adding those structs to `src/discovery/mod.rs` solves it:

```diff
diff --git a/src/discovery/mod.rs b/src/discovery/mod.rs
index 19a6b42..8b0d602 100644
--- a/src/discovery/mod.rs
+++ b/src/discovery/mod.rs
@@ -1,3 +1,6 @@
 pub use self::discovery_manager::DiscoveryManager;
+pub use self::discovery_manager::ServiceInfo;
+pub use self::discovery_manager::ResolveListeners;
+pub use self::discovery_manager::DiscoveryListeners;
 
 pub mod discovery_manager;
```

**Comment by gdesmott** [comment]
Btw it would be nice to add those examples to an `examples` directory.

**Comment by azasypkin** [comment]
Hey @gdesmott, thanks for reporting!

Fixed via https://github.com/fxbox/multicast-dns/commit/f5170d04b150f76e126256fb2f6d73df373c1679 and I've just published the updated version (v0.4.2).
